### DISCUSSION TITLE
Possible performance improvement of `useSharedValue`?

### DISCUSSION THREAD

**Discussion by todorone** [software-mansion/react-native-reanimated#5167] [General]
Either i don't understand something obvious or we can significantly improve `useSharedValue` performance by using `null` as initial value on:
https://github.com/software-mansion/react-native-reanimated/blob/b40463c63e1e399329e9ccdbaa5b14cd9db5716f/src/reanimated2/hook/useSharedValue.ts#L11

We assign initial value if it's `null` on the next lines, so currently we do useless call of `makeMutable(init, oneWayReadsOnly)`(as we discard the result for every call except first one). `makeMutable` function seems do pretty heavy stuff which could be easily avoided improving hook performance and memory consumption.

**Comment by arasrezaei** [comment]
Great topic,
Any benchmarks?

**Comment by tomekzaw** [comment]
See #3199.
